File: server/router/yunguan/product/productComputing.go
package product

import (
	"github.com/gin-gonic/gin"
	v1 "ygpt/server/api/v1"
)

type ProductComputingRouter struct {
}

func (s *ProductComputingRouter) InitProductComputingRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("product/computing")
	supplyInfoApi := v1.ApiGroupApp.YunGuanApiGroup.ProductComputingApi

	{
		dataRouter.POST("addComputingInfo", supplyInfoApi.AddComputingInfo)             // 添加数据
		dataRouter.PUT("updateComputingInfo", supplyInfoApi.UpdateComputingInfo)        // 修改数据
		dataRouter.POST("syncedComputingInfo", supplyInfoApi.SyncedComputingInfo)       // 同步单个数据
		dataRouter.POST("syncedAllComputingInfo", supplyInfoApi.SyncedAllComputingInfo) // 同步所有数据
		dataRouter.DELETE("deleteComputingInfo", supplyInfoApi.DeleteComputingInfo)     // 删除数据
		dataRouter.POST("queryComputingInfo", supplyInfoApi.QueryComputingInfo)         // 获取数据列表
	}
}

File: web/src/api/yunguan/product/productComputing.js
import request from "@/utils/request";

// 添加系统运行工具
export function addComputingInfo(modelData) {
    return request({
        url: "product/computing/addComputingInfo",
        method: "post",
        data: modelData,
    });
}
// 同步单个数据
export function syncedComputingInfo(modelData) {
    return request({
        url: "product/computing/syncedComputingInfo",
        method: "post",
        data: modelData,
    });
}
// 同步所有数据
export function syncedAllComputingInfo() {
    return request({
        url: "product/computing/syncedAllComputingInfo",
        method: "post",
    });
}

// 修改系统运行工具
export function updateComputingInfo(data) {
    return request({
        url: "product/computing/updateComputingInfo",
        method: "put",
        data: data,
    });
}

// 获取系统运行工具列表
export function getComputingInfoList(params) {
    return request({
        url: "product/computing/queryComputingInfo",
        method: "post",
        data: params,
    });
}

// 删除系统运行工具
export function deleteComputingInfo(ID) {
    return request({
        url: "product/computing/deleteComputingInfo",
        method: "delete",
        data: { ID: ID },
    });
}

//同步一条算力产品信息
export function syncComputingInfo(ID) {
    return request({
        url: "product/computing/syncComputingInfo",
        method: "post",
        data: { ID: ID },
    });
}